Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -28,3 +29,71 @@ extension DependencyResolverError: CustomStringConvertible {
         }
     }
 }
+
+public protocol DependencyResolverDelegate {
+    func willResolve(term: Term)
+    func didResolve(term: Term, version: Version)
+
+    func derived(term: Term)
+    func conflict(conflict: Incompatibility)
+    func satisfied(term: Term, by: Assignment, incompatibility: Incompatibility)
+    func partiallySatisfied(term: Term, by: Assignment, incompatibility: Incompatibility, difference: Term)
+    func failedToResolve(incompatibility: Incompatibility)
+    func computed(bindings: [DependencyResolver.Binding])
```

Comment:
^^ are more questionable and mirror the current logging behavior of pub grub resolver